### DISCUSSION TITLE
Specific Spam Reporter

### DISCUSSION THREAD

**Discussion by cochcoder** [ThioJoe/YT-Spammer-Purge#801] [Ideas]
I was on Youtube when I saw a spammer that was never listed when I scanned the video itself. Looking at the channel and their comments I saw that they were writing in a way that would be difficult for the scanner to pickup and their channel name didn't help either. Also, the spammer did the same thing across what seemed to be hundreds of other Youtube channels. I realized that even if I did report this spammer and other future spammers that will use this technique to Youtube and "TJoe.io/filter-feedback" manually. It would still take a lot of time for it to be procced through "TJoe.io/filter-feedback" or just reporting all the comments on Youtube. The feature is already available when you choose specific videos so why not add the option to select it without choosing a specific channel or video to look for the spammer's channel. I understand that it is possible that we might not be able to read all of a channel's comments without looking at other videos; but there might be work arounds for this. For example, if another program was made that would work with "YT-Spammer-Purge" that Youtube channel owners could use to have all the comments that are published on all their videos scanned and logged. So, that when someone scans for a specific spammer's channel/account it could go through the logs/master log and find which channels and videos have comments from that spammer. Also, this method would use much less API quota since if the logs were stored someplace else that doesn't have a API quota, that would cause the API quota to be only used when reporting the comments themselves! I believe that this is a feature that needs to be added so that when the spammers change their technique we will be one step ahead.

**Comment by DinhHuy2010** [comment]
that this so long. can you format it?

**Reply by cochcoder** [reply]
Sure

**Reply by cochcoder** [reply]
It says that besides searching for a specific spammer on one channel and one video at a time when they commented on multiple videos and channels. Why not make a mode where you could scan the spammer's channel and find all their comments across Youtube. And if this isn't something that can be done another program could be made where a Youtuber can have their comments on all their videos scanned and put onto a list that can be scanned for the specific spammer. 

Please keep in mind that this is a very basic simplification of the main post. I recommend that when you can, read the main post since it has some information that could be curial.

**Reply by DinhHuy2010** [reply]
that will hogs up the quota (a.k.a use more quota)

**Reply by cochcoder** [reply]
Normally yes, though if the list of comments are stored on a master log on Github or another place were there is no API quota limit. That would cause the the API quota to be only used when reporting the comments. Also, when it comes to listing the comments for the Youtuber's side it only costs 1 API quota according to [YouTube Data API (v3) - Quota Calculator](https://developers.google.com/youtube/v3/determine_quota_cost) to list the comment. And if you want to make a list of every comment on a channel all you would need is some cleaver scheduling to have it slowly list the comments throughout several days or make multiple Google Cloud projects that all work together causing them to list either 10,000 or 120,000 comments per day on one account depending on how they set it up. Also, the API quota for the reporter's side would cost the same except they only use it to report the spammers. 
![Screenshot (1)](https://user-images.githubusercontent.com/103969142/165558702-ebdb13e3-9a9f-447b-8d37-d897bb01205a.png)

**Reply by DinhHuy2010** [reply]
ok

**Comment by DinhHuy2010** [comment]
but where can get comments from random video?

**Reply by DinhHuy2010** [reply]
or comments id from

**Reply by cochcoder** [reply]
Well to get the comments from a bunch a videos we would need a program/feature that would scan all the comments though instead of reporting them, it copies them all into a list that is public showing the commenter, video they commented, and what they said. Since the program/feature is basically the already present “smart mode”. We can use the code that “smart mode” uses to copy all the comments and comment ids into a local log. But instead we change it to upload the comments.

**Comment by Tranberg666** [comment]
When a bunch of bots are posting the same irrelevant video link comment under all popular comments on certain popular channels, why an I not allowed to report them all, when I find them with filter search?

Specifically "Finally its here" and a link to a baking channel 

This is what they write: " https://youtu.be/93Fyv4XAr-A Finally its here."

**Reply by Firecul** [reply]
Your issue #817 will hopefully inspire ThioJoe to add that phrase and link pattern to the filters
